File: c/Algorithms/sorts/testcases/test_sorts.c
#include <sort_common.h>
#include <assert.h>

int main()
{
	void (*sortfn) (int a[], int n) = NULL;
	int *a, n;
#ifdef _SELECTION_SORT_
	sortfn = selection_sort;
#elif _BUBBLE_SORT_
	sortfn = bubble_sort;
#elif _INSERTION_SORT_
	sortfn = insertion_sort;
#elif _MERGE_SORT_
	sortfn = merge_sort;
#elif _QUICK_SORT_
	sortfn = quick_sort;
#elif _BUBBLE_SORTR_
	sortfn = bubble_sortR;
#endif

	assert(sortfn != NULL);

	printf("n: ");
	scanf("%d", &n);
	
	getArray(&a, n);
	
	sortfn(a, n);
	
	printArray(a, n);

	return 0;
}

File: python/algorithms/sorts/testcases/test_sorts.py
from algorithms.sorts import *
import random

def test_sortfn(sortfn):
	a = [5,4,3,2,1]
	assert sortfn(a) == [1,2,3,4,5]
	x = [random.randint(0, 1000) for _ in xrange(100)]
	assert sortfn(x) == sorted(x)


if __name__ == '__main__':
	test_sortfn(abacus_sort)
	test_sortfn(merge_sort)
	test_sortfn(heap_sort)
	test_sortfn(insertion_sort)
	test_sortfn(comparison_counting_sort)
	# FIXME: Fix infinite loop in quicksort/partition
	# fails for random inputs tc
	# test_sortfn(quick_sort)
	test_sortfn(counting_sort)
	test_sortfn(radix_sort)

